controller/lib/comms/comms_nanopb.cpp

```diff
@@ -41,8 +38,7 @@ void comms_sendPeriodicReadings(float pressure, float volume, float flow) {
   if (output_buffer_ready) {
     return;
   }
-  uint64_t time;
-  time = millis();
+  uint64_t time = millis();
```

Comment:
millis() returned by arduino is a uint32_t.

Reply:
But we will have to use uint64 for millis, as 49 days (2**32/1000/3600/24) feels a bit unnerving? 

Probability of somebody leaving the machine on for 7 weeks and and then deciding to use it? Or not switching it off at all during use? Dunno, maybe I'm just over paranoid.

Reply:
I agree that requiring a restart every 7 weeks is unnerving.  I am reminded of http://www-users.math.umn.edu/~arnold//disasters/patriot.html

We could try to handle overflow correctly everywhere, but it's basically impossible to test, and it's hard to get right.

We could assume that Hal::millis() will be called at least once every 7 weeks and detect when time reported by `::millis()` goes backwards.  Then `Hal.millis()` (which is what we should be calling here instead of `::millis()`) can be `uint64_t`?

Reply:
It wouldn't go negative as any sane person would interpret the number as unsigned, I hope :)
Question is what happens on rollover?
Yes, I propose to go 64bits on `Hal.millis()`